hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/upsert/GenericUpsertQueryGenerator.java

```diff
@@ -77,6 +77,9 @@ public String getUpsertQuery() {
 
         // {0} is column name and {1} is column default. t or blank is the temporary table alias and e is the existing.
         velocityContext.put("coalesceColumns", metadata.columns("coalesce({0}, e_{0}, {1})"));
+        velocityContext.put(
+                "coalesceHistoryColumns",
+                metadata.columns("coalesce({0}, e_{0}, {1})", c -> !c.isUpdatable(), "coalesce(e_{0}, {0}, {1})"));
```

Comment:
This syntax is a bit wonky with two patterns separated by a predicate. From the signature it's not clear how this works in contrast to the existing method and it's not clear which predicate value uses what pattern. I think it would be cleaner if we reuse the existing method and create two separate lists that we combine:

```suggestion
        var coalesceHistoryColumns = metadata.colummns("coalesce({0}, e_{0}, {1})", EntityMetadata::isUpdatable);
        coalesceHistoryColumns += "," + metadata.colummns("coalesce(e_{0}, {0}, {1})", c -> !c.isUpdatable());

        // {0} is column name and {1} is column default. t or blank is the temporary table alias and e is the existing.
        velocityContext.put("coalesceColumns", metadata.columns("coalesce({0}, e_{0}, {1})"));
        velocityContext.put("coalesceHistoryColumns", coalesceHistoryColumns));
```

Reply:
I investigated this. The statement that is used by the upsert sql is: 
```
  insert into $historyTable ($insertColumns)
  select distinct $coalesceHistoryColumns
  ```
  
  So the column order of the `insertColumns` and the `coalesceHistoryColumns` have to match, and concatenating the two calls to `columns` for `coalesceHistoryColumns` results in a different column ordering. 
  
  Another solution might be to change the newly added `columns` method and signature to something more descriptive like `columnsOrElse(Predicate<ColumnMetadata> predicate, String predicatePattern, String elsePattern)`

Reply:
Ok, if it doesn't work then we can leave as is.